src/VisualStudio/Core/Test/ClassView/SyncClassViewTests.vb

```diff
@@ -0,0 +1,661 @@
+        Public Async Function VisualBasic_TestMethod3() As Task
+            Dim workspace =
+<Workspace>
+    <Project Language="Visual Basic" CommonReferences="true" AssemblyName="VBTestAssembly">
+        <Document>
+            Namespace N
+                Class C
+                    Sub $$M()
+                    End Sub
+                End Class
+            End Namespace
+        </Document>
+    </Project>
+</Workspace>
+
+            Await TestAsync(workspace,
+                 canonicalNodes:={
+                    Package("VBTestAssembly"),
+                    [Namespace]("N"),
+                    [Class]("C"),
+                    Member("M()")
+                 },
+                 presentationNodes:={
+                    Package("VBTestAssembly"),
+                    [Namespace]("N"),
+                    [Class]("C"),
+                    Member("M()")
+                 })
+        End Function
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.VsNavInfo)>
+        Public Async Function VisualBasic_TestMethod4() As Task
+            Dim workspace =
+<Workspace>
+    <Project Language="Visual Basic" CommonReferences="true" AssemblyName="VBTestAssembly">
+        <Document>
+            Namespace N
+                Class C
+                    $$Sub M()
+                    End Sub
+                End Class
+            End Namespace
+        </Document>
+    </Project>
+</Workspace>
+
+            Await TestAsync(workspace,
+                 canonicalNodes:={
+                    Package("VBTestAssembly"),
+                    [Namespace]("N"),
+                    [Class]("C"),
+                    Member("M()")
+                 },
+                 presentationNodes:={
+                    Package("VBTestAssembly"),
+                    [Namespace]("N"),
+                    [Class]("C"),
+                    Member("M()")
+                 })
+        End Function
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.VsNavInfo)>
+        Public Async Function VisualBasic_TestMethod5() As Task
+            Dim workspace =
+<Workspace>
+    <Project Language="Visual Basic" CommonReferences="true" AssemblyName="VBTestAssembly">
+        <Document>
+            Namespace N
+                Class C
+            $$        Sub M()
+                    End Sub
+                End Class
+            End Namespace
+        </Document>
+    </Project>
+</Workspace>
+
+            Await TestAsync(workspace,
+                 canonicalNodes:={
+                    Package("VBTestAssembly"),
+                    [Namespace]("N"),
+                    [Class]("C"),
+                    Member("M()")
+                 },
+                 presentationNodes:={
+                    Package("VBTestAssembly"),
+                    [Namespace]("N"),
+                    [Class]("C"),
+                    Member("M()")
+                 })
+        End Function
+
```

Comment:
Some tests for fields/events/properties?

Negative tests for things that aren't valid symbols (operators)?

Negative test when outside any named type?